### DISCUSSION TITLE
How to integrate CSS and SVG for mall style floor map?

### DISCUSSION THREAD

**Discussion by michcfr** [maplibre/maplibre-gl-js#4293] [Ideas]
Hello,
I would like to use MapLibre GL JS for 3D indoor mapping but not with polygon extrusion. 
I rather prefer using mall-style floormap (based on CSS and SVG) like that: https://tympanus.net/codrops/2016/04/13/interactive-3d-mall-map/

Is it possible? Any hints on how to do the integration?

thank you

**Comment by HarelM** [comment]
You can use image source.
I think this question is similar to the following one:
https://github.com/maplibre/maplibre-gl-js/issues/4175

You can convert the svg to "raster" image and use that.
I would recommend converting it to a geojson and style it using maplibre style spec.
